src/dune_engine/forker.mli

```diff
@@ -0,0 +1,36 @@
+open Import
+
+module Foreman : sig
```

Comment:
Could you explain the need for the "foreman"? 

Dune can start all action runners during its own initialisation, so it knows about them and can therefore dispatch action execution requests to each directly. Why do we need an intermediate?

Reply:
Maybe the name is poor? Foreman is just an API for managing runners that is independent of the command line interface and the algorithm used to distribute action across runners.

> Dune can start all action runners during its own initialisation

The intent of Foreman module is to provide an API to help with exactly that.

Reply:
To answer this more directly:

> Why do we need an intermediate?

* A foreman reaps all the runners to avoid zombies
* A foreman handles distributing actions to workers that haven't yet started by managing an ivar that will be filled when the initialization is complete
* A foreman hides all the RPC related implementation details. So if we choose a different mechanism, modify the versioning scheme, or the serialization format, all of those changes would only need to occur in this one module.

Reply:
Thanks! This does clear things up.